stan/math/prim/scal/prob/skew_normal_rng.hpp

```diff
@@ -1,39 +1,79 @@
-#include <stan/math/prim/scal/meta/include_summand.hpp>
+#include <stan/math/prim/scal/meta/scalar_seq_view.hpp>
+#include <stan/math/prim/scal/meta/VectorBuilder.hpp>
+#include <stan/math/prim/scal/prob/normal_rng.hpp>
+#include <boost/random/normal_distribution.hpp>
+#include <boost/random/variate_generator.hpp>
 #include <string>
 
 namespace stan {
   namespace math {
-
-    template <class RNG>
-    inline double
-    skew_normal_rng(double mu,
-                    double sigma,
-                    double alpha,
+    /**
+     * Return a pseudorandom Skew-normal variate for the given location, scale,
+     * and shape using the specified random number generator.
+     *
+     * mu, sigma, and alpha can each be either scalars or vectors. All vector
+     * inputs must be the same length.
+     *
+     * @tparam T_loc Type of location parameter
+     * @tparam T_scale Type of scale parameter
+     * @tparam T_shape Type of shape parameter
+     * @tparam RNG type of random number generator
+     * @param mu (Sequence of) location parameter(s)
+     * @param sigma (Sequence of) scale parameter(s)
+     * @param alpha (Sequence of) shape parameter(s)
+     * @param rng random number generator
+     * @return Skew-normal random variate
+     * @throw std::domain_error if mu is infinite, sigma is nonpositive, or
+     * alpha is infinite
+     * @throw std::invalid_argument if vector arguments are not the same length
+     */
+    template <typename T_loc, typename T_scale, typename T_shape, class RNG>
+    inline typename VectorBuilder<true, double, T_loc, T_scale, T_shape>::type
+    skew_normal_rng(const T_loc& mu,
+                    const T_scale& sigma,
+                    const T_shape& alpha,
                     RNG& rng) {
-      boost::math::skew_normal_distribution<> dist(mu, sigma, alpha);
+      using boost::variate_generator;
+      using boost::random::normal_distribution;
 
       static const std::string function = "skew_normal_rng";
 
       check_finite(function, "Location parameter", mu);
+      check_positive_finite(function, "Scale parameter", sigma);
       check_finite(function, "Shape parameter", alpha);
-      check_positive(function, "Scale parameter", sigma);
+      check_consistent_sizes(function,
+                             "Location parameter", mu,
+                             "Scale Parameter", sigma,
+                             "Shape Parameter", alpha);
 
-      return quantile(dist, uniform_rng(0.0, 1.0, rng));
-    }
+      scalar_seq_view<T_loc> mu_vec(mu);
+      scalar_seq_view<T_scale> sigma_vec(sigma);
+      scalar_seq_view<T_shape> alpha_vec(alpha);
 
+      size_t N = max_size(mu, sigma, alpha);
+
+      VectorBuilder<true, double, T_loc, T_scale, T_shape> output(N);
+
+      variate_generator<RNG&, normal_distribution<> >
+        norm_rng(rng, normal_distribution<>(0.0, 1.0));
+      for (size_t n = 0; n < N; n++) {
+        double r1 = norm_rng(),
+          r2 = norm_rng();
```

Comment:
[optional]

I'd prefer to have one declaration per line.  And of course, tighten up vertical spaces!